### PR TITLE
Add minip volume rendering

### PR DESCRIPTION
**Pull request by kevinyamauchi**
As discussed in #86 , this PR adds minimum intensity projection rendering. 

https://github.com/pygfx/pygfx/assets/1120672/b289208c-035d-4f5d-b551-743613fd1d82

^ example script: `/examples/feature_demo/volume_minip_render.py`

### CODE REVIEW

**Comment by kevinyamauchi** [comment]
I believe the basic functionality is working and I would appreciate feedback on the following :
- what do we want to call this rendering mode? As a starting point I chose `minip`, but I am open to other suggestions.
- there is some code duplicated between the minip a mip `raycast()` functions. do we want to refactor so there is less duplication (e.g., the coloring and picking code)? I think the functions are short enough that it is fine as is, but I am fine with refactoring if you prefer.

Thanks!

File: pygfx/renderers/wgpu/wgsl/volume_ray.wgsl [hunk 1]
```diff
@@ -213,6 +213,75 @@ $$ if mode == 'mip'
         return out;
     }
 
+$$ elif mode == 'minip'
+    // raycasting function for MIP rendering.
```
Comment (almarklein) [inline pygfx/renderers/wgpu/wgsl/volume_ray.wgsl:217]:
```suggestion
    // raycasting function for MINimum Intensity Projection rendering.
```

**Review by almarklein — COMMENTED** [review]
>  what do we want to call this rendering mode? As a starting point I chose minip, but I am open to other suggestions.

The name "minip" seems to be widely used. It's what vispy uses and e.g. https://en.wikipedia.org/wiki/Minimum_intensity_projection. So let's just use that.

> I think the functions are short enough that it is fine as is

I agree.

**Review by almarklein — APPROVED** [review]

**Review by Korijn — APPROVED** [review]

**Comment by almarklein** [comment]
Thanks! 🙏

**Pull request merged**